src/main/scala/edu/cmu/cs/obsidian/codegen/CodeGenYul.scala

```diff
@@ -186,7 +196,7 @@ object CodeGenYul extends CodeGenerator {
             transaction.name, // TODO rename transaction name (by adding prefix/suffix)
             transaction.args.map(v => TypedName(v.varName, mapObsTypeToABI(v.typIn.toString))),
             ret,
-            Block(transaction.body.flatMap(translateStatement))))
+            Block(transaction.body.flatMap((s: Statement) => translateStatement(s))))) //todo iev temp vars: likely a hack to work around something wrong
```

Comment:
What is this todo for?

Reply:
in general i'm not sure when i'm using `flatMap` because it's the right thing versus hiding a bug in how i'm thinking about statements, so i have some todos just as bookmarks to help me find places that i don't quite trust.